Trad carol (#13)
appliedepi/epirhandbook_pt: new_pages/ggplot_basics.Rmd

# O básico o ggplot{}

```{r, out.width=c('100%', '100%'), fig.align = "center", echo=F}
knitr::include_graphics(here::here("images", "ggplot_basics_top.png"))
```

O **ggplot2** é o pacote R mais popular de visualização de dados. Sua função `ggplot()` está no centro deste pacote, e toda esta abordagem é conhecida coloquialmente como *"ggplot "* com as imagens resultantes às vezes carinhosamente chamadas de "ggplots". O "gg" nestes nomes reflete a "**g**ramática de **g**ráficos" utilizados para construir as figuras. O **ggplot2** beneficia-se de uma grande variedade de pacotes R suplementares que melhoram ainda mais sua funcionalidade.

> A sintaxe é significativamente diferente dos gráficos gerados pelo R **base***, e tem uma curva de aprendizado associada a ela. A utilização do **ggplot2** geralmente exige que o usuário formate seus dados de uma forma altamente **compatível com o "tidyverse**, o que em última análise torna a utilização conjunta destes pacotes muito eficaz.

Author: A sintaxe é significativamente diferente dos gráficos gerados pelo R **base***, e tem uma curva de aprendizado associada a ela. A utilização do **ggplot2** geralmente exige que o usuário formate seus dados de uma forma altamente **compatível com o "tidyverse"**, o que em última análise torna a utilização conjunta destes pacotes muito eficaz.

> A sintaxe é significativamente diferente dos gráficos gerados pelo R **base***, e tem uma curva de aprendizado associada a ela. A utilização do **ggplot2** geralmente exige que o usuário formate seus dados de uma forma altamente **compatível com o "tidyverse**, o que em última análise torna a utilização conjunta destes pacotes muito eficaz.

Author: ```suggestion
A sintaxe é significativamente diferente dos gráficos gerados pelo R **base***, e tem uma curva de aprendizado associada a ela. A utilização do **ggplot2** geralmente exige que o usuário formate seus dados de uma forma altamente **compatível com o "tidyverse"**, o que em última análise torna a utilização conjunta destes pacotes muito eficaz.

```

  theme()                                 # aqui você pode ajustar a cor, fonte, tamanho, de elementos que não são orindos dos dados (eixo, título, etc.) 
```

 

## `ggplot()`  

O comando de abertura de qualquer gráfico ggplot2 é `ggplot()`. Este comando simplesmente cria uma tela em branco sobre a qual se pode adicionar camadas. Ele "abre" o caminho para que outras camadas sejam adicionadas com um símbolo `+`.

Normalmente, o comando `ggplot()` inclui o `dados = ` argumento para o gráfico. Isto define o conjunto de dados padrão a ser utilizado para as camadas subsequentes do gráfico.  

Este comando terminará com um `+` após seus parênteses de fechamento. Isto deixa o comando "aberto". O ggplot só será executado/aparecerá quando o comando completo incluir uma camada final *sem* um `+` no final.  

```{r, eval=F}
# Isso irá criar um gráfco que é um painel em branco 
ggplot(data = linelist)
```

## Geoms  

Uma tela em branco certamente não é suficiente - precisamos criar geometrias (formas) a partir de nossos dados (por exemplo, gráficos de barra, histogramas, gráficos de dispersão, gráficos de caixa).  

> Isto é feito adicionando camadas "geoms" ao comando inicial `ggplot()`. Há muitas funções do **ggplot2**  que criam "geoms". Cada uma destas funções começa com "geom_", portanto, vamos nos referir a elas genericamente como `geom_XXXX()`. Há mais de 40 geoms em **ggplot2** e muitos outros criados por ventiladores. Veja-os na [galeria ggplot2](https://exts.ggplot2.tidyverse.org/gallery/). Alguns geoms comuns estão listados abaixo:  

Author: Acho que vou deixar ventiladores nesse aqui mesmo... q achei mais adequado. @halian-vilela

* No segundo exemplo, o histograma requer apenas o eixo x mapeado para uma coluna. O histograma `binwidth = `, `color = `, `fill = ` (cor de preenchimento), e `alpha = ` ` são novamente definidos dentro do geom para valores estáticos.  

```{r, out.width=c('50%', '50%'), fig.show='hold', warning=F, message=F}
# gráfico de dispersão
ggplot(data = linelist, mapping = aes(x = age, y = wt_kg))+  # configurar mapeamento dos dados e eixos 
  geom_point(color = "darkgreen", size = 0.5, alpha = 0.2)         # configurar a estética estática dos pontos

# histograma
ggplot(data = linelist, mapping = aes(x = age))+       #onfigurar mapeamento dos dados e eixos 
  geom_histogram(              # mostra o histograma
    binwidth = 7,                # largura das classes
    color = "red",               # cor da linha da classe
    fill = "blue",               # cor de preenchimento da barra
    alpha = 0.1)                 #transparência da barra
```

### Responsivo a valores de uma coluna {.unnumbered}  

> A alternativa é mapear os argumentos da estética de um gráfico a uma coluna (variável) do seus dados. Nesta abordagem, a exibição deste *aesthetics* dependerá do valor desta observação naquela coluna dos dados. Se os valores da coluna forem contínuos, a escala de exibição (legenda) para aquele *aesthetics* será contínua. Se os valores da coluna forem discretos, a legenda exibirá cada valor e os dados plotados aparecerão como claramente "agrupados" (leia mais na seção [agrupamento](#ggplotgroups) desta página).  

Reviewer: ```suggestion
A alternativa é mapear os argumentos da estética de um gráfico a uma coluna (variável) do seus dados. Nesta abordagem, a exibição deste *aesthetics* dependerá do valor desta observação naquela coluna dos dados. Se os valores da coluna forem contínuos, a escala de exibição (legenda) para aquele *aesthetics* será contínua. Se os valores da coluna forem discretos, a legenda exibirá cada valor e os dados plotados aparecerão "agrupados" de forma distinta (leia mais na seção [agrupamento](#ggplotgroups) desta página).  

```

A ideia do "distinctly" aqui é mais ao pé da letra mesmo... o sentido é dizer que o agrupamento será feito de forma a criar uma "distinção" entre os grupos de dados.

Para conseguir isso, você mapeia esse *aesthetics* do gráfico para um nome de *coluna* (não entre aspas). Isto deve ser feito *em uma função `mapping = aes()` (nota: há vários lugares no código que você pode fazer estas atribuições de mapeamento, como discutido [abaixo](##ggplot_basics_map_loc))).  

Dois exemplos estão abaixo.  

* No primeiro exemplo, a *aesthetics* `color = ` (de cada ponto) é mapeada para a coluna `age` - e uma escala de cores apareceu na legenda! Por enquanto, basta observar que a escala existe - mostraremos como modificá-la em seções posteriores.  

> * No segundo exemplo, duas novas *aesthetics* do gráfico também são mapeados para colunas (`color = ` e `size = `), enquanto a *aesthetics* do gráfico `shape = ` e `alpha = ` são mapeados para valores estáticos fora de qualquer `mapping = aes()` função.   

Reviewer: ```suggestion
* No segundo exemplo, duas novas *aesthetics* do gráfico também são mapeados para colunas (`color = ` e `size = `), enquanto a *aesthetics* do gráfico `shape = ` e `alpha = ` são mapeados para valores estáticos fora de qualquer função `mapping = aes()`.   

```

```{r, out.width=c('50%', '50%'), fig.show='hold', warning=F, message=F}
# gráfico de dispersão

ggplot(data = linelist,   # escolha os dados
       mapping = aes(     # mapeie as aesthetics para coluna de valores

>          x = age,           # mapeie o eixo x pata idade (coluna age)       

Reviewer: ```suggestion
         x = age,           # mapeie o eixo x para idade (coluna age)       

```

>          y = wt_kg,         # mapeie o eixo y pata peso (wt_kg)      

Reviewer: ```suggestion
         y = wt_kg,         # mapeie o eixo y para peso (wt_kg)      

```

>          color = age) # mapeie a cor segundo a iddade

Reviewer: ```suggestion
         color = age) # mapeie a cor segundo a idade

```

       )+     
  geom_point()         # mostre como pontos

# gráfico de dispersão
ggplot(data = linelist,   # escolha os dados
       mapping = aes(     # mapeie as aesthetics para coluna de valores

>          x = age,           # mapeie o eixo x pata idade (coluna age)    

Reviewer: ```suggestion
         x = age,           # mapeie o eixo x para idade (coluna age)    

```

>          y = wt_kg,         # mapeie o eixo y pata peso (wt_kg) 

Reviewer: ```suggestion
         y = wt_kg,         # mapeie o eixo y para peso (wt_kg) 

```

>          color = age,       # mapeie a cor segundo a iddade

Reviewer: ```suggestion
         color = age,       # mapeie a cor segundo a idade

```

>          size = age))+      # mapeie o tamanho segundo a iddade

Reviewer: ```suggestion
         size = age))+      # mapeie o tamanho segundo a idade

```

  geom_point(             # mostre como pontos
    shape = "diamond",      # pontos como losangos (diamantes)

>     alpha = 0.3)            # tranparência do ponto a 30%

Reviewer: ```suggestion
    alpha = 0.3)            # transparência do ponto a 30%

```

```

Nota: Os argumentos referentes aos eixos são sempre atribuídos a colunas dos dados (não a valores estáticos), e isto é sempre feito dentro de `mapping = aes()`.  

> Torna-se importante acompanhar as camadas e a *aesthetics* de seus lotes ao fazer lotes mais complexos - por exemplo, lotes com múltiplos *geoms*. No exemplo abaixo, a aesthetic `size = ` eé atribuída duas vezes - uma para `geom_point()` e outra para `geom_smooth()` - ambas as vezes como um valor estático.   

Reviewer: ```suggestion
Torna-se importante acompanhar as camadas e a *aesthetics* de seus gráficos à medida que vão ficando mais complexos - por exemplo, gráficos com múltiplos *geoms*. No exemplo abaixo, a aesthetic `size = ` é atribuída duas vezes - uma para `geom_point()` e outra para `geom_smooth()` - ambas as vezes como um valor estático.   

```

```{r, warning=F, message=F}
ggplot(data = linelist,
       mapping = aes(           # mapeia a aesthetics para as colunas
         x = age,
         y = wt_kg,
         color = age_years)
       ) + 
  geom_point(                   # addiciona pontos para cada linha de dados
    size = 1,
    alpha = 0.5) +  

>   geom_smooth(                  # adiciona uma linha de tendencia 

Reviewer: ```suggestion
  geom_smooth(                  # adiciona uma linha de tendência 

```

    method = "lm",              # com um método linear

>     size = 2)                   # tmanho (grossura da linha) de 2

Reviewer: ```suggestion
    size = 2)                   # tamanho (grossura da linha) de 2

```

```

> ### Onde incluir os atributos de mapeamento {#ggplot_basics_map_loc .unnumbered}

Reviewer: Parei aqui... 😄

Author: Top, assim q eu terminar de revisar o de limpeza  de dados eu posso continuar aqui tb

Ao utilizar `facet_wrap()` ou `facet_grid()`, podemos adicionar `escalas = "free_y"` a "free" ou liberar os eixos y dos painéis para escalar adequadamente seu subconjunto de dados. Isto é particularmente útil se as contagens reais forem pequenas para uma das subcategorias e se as tendências forem difíceis de se observar de outra forma. Em vez de "free_y" também podemos escrever "free_x" para fazer o mesmo para o eixo x (por exemplo, para datas) ou "free" para ambos os eixos. Note que em `facet_grid', a escala y será a mesma para facetas na mesma linha, e a escala x será a mesma para facetas na mesma coluna.

Ao utilizar somente `facet_grid`, podemos adicionar `space = "free_y"` ou `space = "free_x"` para que a altura ou largura real da faceta seja ponderada para os valores da figura dentro. Isto só funciona se `scales = "free"` (y ou x) já estiver aplicado. 

```{r, message=FALSE, warning=FALSE}

# Eixos livres

ggplot(malaria_data, aes(x = data_date, y = malaria_tot)) +
  geom_col(width = 1, fill = "darkred") +       # plote os dados de contagem como colunas
  theme_minimal()+                              # simplifique os painel de fundo

>   labs(                                         #adicione rótulos, título, etc.

Author: essa foi a correção mais importante @halian-vilela

>   labs(                                         #adicione rótulos, título, etc.

Author: ^^' hehehe

## Rótulos

Certamente você vai querer adicionar ou ajustar os rótulos (elementos de texto) do gráfico. Estes ajustes são mais facilmente feitas dentro da função `labs()` que é adicionada à trama com `+` tal como os geoms eram.  

Dentro de `labs()` você pode fornecer *strings*  para estes parâmetros:  

* `x = ` e `y = ` O título do eixo x e do eixo y (etiquetas)  
* `title = `  O título principal do gráfico
* `subtitle = `  O subtítulo do gráfico, em texto menor abaixo do título  
* `caption =` A legenda do gráfico, na parte inferior direita por padrão  

Aqui está um gráfico que fizemos anteriormente, mas com rótulos mais bonitos:  

```{r, warning=F, message=F}
age_by_wt <- ggplot(
  data = linelist,   # especifique o objeto com os dados
  mapping = aes(     # mapeie as aesthetics para colunas desses dados
         x = age,           # mapeie o eixo x para idade (coluna age)            
         y = wt_kg,         # mapeie o eixo x para peso (coluna wt_kg)       
         color = age))+     # mapeie a cor para a idade
  geom_point()+           # mostre os dados como pontos
  labs(

>     title = "Distribioção de idade e peso",

Reviewer: ```suggestion
    title = "Distribuição de idade e peso",

```

    subtitle = "Surto de Ebola fictício, 2014",
    x = "Idade em anos",
    y = "Peso em quilos",
    color = "Idade",
    caption = stringr::str_glue("Dados como em {max(linelist$date_hospitalisation, na.rm=T)}"))

age_by_wt
```

> Observe como na atribuição do título utilizamos `str_glue()` do pacote **stringr** para implantar código R dinâmico dentro do texto da string. O legenda mostrará os "Dados a partir de": "data que reflete a data máxima de hospitalização na lista de discussão. Leia mais sobre isto na página em [Caracteres e strings].  

Reviewer: ```suggestion
Observe como na atribuição do título utilizamos `str_glue()` do pacote **stringr** para implantar código R dinâmico dentro do texto da string. O legenda mostrará os "Dados a partir de": "data que reflete a data máxima de hospitalização na linelist". Leia mais sobre isto na página em [Caracteres e strings].  

```

Uma nota sobre a especificação do título da  *legenda*: Não há nenhum argumento de "título da legenda", pois você poderia ter múltiplas escalas em sua legenda. Dentro de `labs()`, você pode escrever o argumento da  *aesthetic* do gráfico utilizada para criar a legenda, e fornecer o título desta forma. Por exemplo, acima atribuímos `color = age` para criar a legenda. Portanto, fornecemos `color = ` a `labs()` e atribuímos o título da legenda desejada ("Idade" com I maiúsculo). Se você criar a legenda com `aes(fill = COLUMN)`, então em `labs()` você escreveria `fill = ` para ajustar o título dessa legenda. A seção sobre escalas de cores na página [Dicas para o ggplot] fornece mais detalhes sobre edição de legendas, e uma abordagem alternativa utilizando funções `scales_()`. 

> ## Temas  {#ggplot_basics_themes} 

Reviewer: **Parei aqui.** 🕜

> # O básico o ggplot{}

Author: ```suggestion
# O básico do ggplot{}

```

## Temas  {#ggplot_basics_themes} 

Uma das melhores partes do **ggplot2** é a quantidade de controle que você tem sobre o gráfico - você pode definir qualquer coisa! Como mencionado acima, o desenho do gráfico que *não* está relacionado às formas/geometrias dos dados são ajustados dentro da função `theme()`. Por exemplo, a cor de fundo do gráfico, presença/ausência de linhas de grade e a fonte/tamanho/cor/alinhamento do texto (títulos, subtítulos, legendas, texto do eixo...). Estes ajustes podem ser feitos de uma de duas maneiras:  

> *Adicionar um [*tema completo*](https://ggplot2.tidyverse.org/reference/ggtheme.html) `theme_()`função para fazer ajustes de varredura - estes incluem `theme_classic()`, `theme_minimal()`, `theme_escuro()`, `theme_light()`theme_grey()`, `theme_bw()` entre outros  

Author: ```suggestion
*Adicionar um [*tema completo*](https://ggplot2.tidyverse.org/reference/ggtheme.html) `theme_()`função para fazer ajustes de varredura - estes incluem `theme_classic()`, `theme_minimal()`, `theme_dark()`, `theme_light()`theme_grey()`, `theme_bw()` entre outros  

```

* Começamos com o gráfico `age_by_wt` definido logo acima e acrescentamos `theme_classic()`  
* Para ajustes mais finos, adicionamos `theme()` e incluímos um argumento para cada elemento do gráfico a ser ajustado  

Pode ser bom organizar os argumentos em seções lógicas. Para descrever apenas alguns dos utilizados abaixo:  

* `legend.position = ` é único no sentido de aceitar valores simples como "bottom", "top", "left", e "right". Mas geralmente, os argumentos relacionados ao texto exigem que você coloque os detalhes *no* `element_text()`.  
* tamanho do título com `element_text(size = 30)`.  
* O alinhamento horizontal do título com `element_text(hjust = 0)` (da direita para a esquerda)  
* O subtítulo está em itálico com `element_text(face = "italic")`  

```{r, , warning=F, message=F}
age_by_wt + 
  theme_classic()+                                 # ajustes de tema pré-definidos
  theme(
    legend.position = "bottom",                    # move a legenda  para baixo
    
    plot.title = element_text(size = 30),          # ajusta o tamanho do título para 30
    plot.caption = element_text(hjust = 0),        # alinha a Legenda da figura a esquerda

>     plot.subtitle = element_text(face = "italic"), # italicize subtitle

Author: ```suggestion
    plot.subtitle = element_text(face = "italic"), # deixa subtítulo em itálico

```

    
    axis.text.x = element_text(color = "red", size = 15, angle = 90), # ajusta o texto o eixo x 
    axis.text.y = element_text(size = 15),         # ajusta o texto o eixo y
    
    axis.title = element_text(size = 20)           # ajusta o texto de ambos os eixos
    )     
```

Aqui estão alguns argumentos especialmente comuns em `theme()`. Você reconhecerá alguns padrões, tais como acrescentar `.x' ou `.y' para aplicar a mudança somente em um eixo.  

Argumento `theme()`                |O que é ajustado
-----------------------------------|----------------------------------
`plot.title = element_text()`      |O título
`plot.subtitle = element_text()`   |O subtítulo
`plot.caption = element_text()`    |A legenda da figura (parâmetros: family, face, color, size, angle,...) 
`axis.title = element_text()`      |Títulos e ambos eixos (ambos x e y) (parâmetros: size, face, color...)
`axis.title.x = element_text()`    |Título do eixo x (use `.y` para apenas eixo Y)
`axis.text = element_text()`       |Texto do eixo x (ambos x and y)

> `axis.text.x = element_text()`     |Axis text x-axis only (use `.y` para apenas eixo Y)  

Author: ```suggestion
`axis.text.x = element_text()`     | Texto do eixo x (use `.y` para apenas eixo Y)  

```

`strip.background = element_rect()`|Rótulo da faceta (parâmetros: fill, colour, size...)  

Mas há tantos argumentos para temas! Como eu poderia me lembrar de todos eles? Não se preocupe - é impossível lembrar-se de todos eles. Felizmente, existem algumas ferramentas para ajudá-lo:  

A documentação sobre [modificação do tema](https://ggplot2.tidyverse.org/reference/theme.html), que tem uma lista completa.   

<span style="color: darkgreen;">**_DICA:_** Rode `theme_get()` do **ggplot2** para listar os mais de 90 argumentos da função `theme()` no console .</span>  

<span style="color: darkgreen;">**_DICA:_** Se você em algum momento quiser remover um elemento do gráfico, você também o pode fazer com o `theme()`. Apenas passe o parâmetro `element_blank()` para um argumento para que ele desapareca completamente. Para legendas, configure `legend.position = "none".`</span>  

## Cores 

Por favor veja [a seção sobre escala de cores na página de dicas do ggplot](#ggplot_tips_colors).  

## Usando o pipe (%>%) no **ggplot2**   

Ao utilizar *pipes* (%>%) para limpar e transformar seus dados, é fácil passar os dados transformados em `ggplot()`.  

> Os *pipes* que passam o conjunto de dados de função para função passarão a ser um símbolo de mais `+` assim que a função `ggplot()` for chamada. Observe que, neste caso, não há necessidade de especificar o argumento `dados = `, pois este é automaticamente definido como o conjunto de dados que vinha sendo usado no *pipe*.  

Author: ```suggestion
Os *pipes* que passam o conjunto de dados de função para função passarão a ser um símbolo de mais `+` assim que a função `ggplot()` for chamada. Observe que, neste caso, não há necessidade de especificar o argumento `data = ` (data), pois este é automaticamente definido como o conjunto de dados que vinha sendo usado no *pipe*.  

```

Esta é a aparência que pode ter:  

```{r, warning=F, message=F}
linelist %>%                                                     # comece com a linelist
  select(c(case_id, fever, chills, cough, aches, vomit)) %>%     # selecione as colunas
  pivot_longer(                                                  # faça o pivotamento para ficar mais longo
    cols = -case_id,                                  
    names_to = "nome_sintoma",
    values_to = "sintoma_esta_presente") %>%
  mutate(                                                        # substitua valores faltantes
    sintoma_esta_presente = replace_na(sintoma_esta_presente, "desconhecido")) %>% 
  
  ggplot(                                                        # comece o gráfico!
    mapping = aes(x = nome_sintoma, fill = sintoma_esta_presente))+
  geom_bar(position = "fill", col = "black") +                    
  theme_classic() +
  labs(
    x = "Sintoma",
    y = "Status do sintoma (proporção)"
  )
```

## Fazer gráficos de dados contínuos

> Ao longo desta página, você já viu muitos exemplos de plotagem de dados contínuos. Aqui nós os consolidamos brevemente e apresentamos algumas variações.  

Author: ```suggestion
Ao longo desta página, você já viu muitos exemplos de gráficos de dados contínuos. Aqui nós os consolidamos brevemente e apresentamos algumas variações.  

```

```{r fig.show='hold', message=FALSE, warning=FALSE, out.width=c('50%', '50%')}
# A) Boxplot geral
ggplot(data = linelist)+  
  geom_boxplot(mapping = aes(y = age))+   # apenas o y é mapeado
  labs(title = "A) Boxplot geral")

# B) Box plot por grupo
ggplot(data = linelist, mapping = aes(y = age, x = gender, fill = gender)) + 
  geom_boxplot()+                     
  theme(legend.position = "none")+   # remove a legenda
  labs(title = "B) Box plot por sexo")      
```

Para obter o código para adicionar um boxplot às bordas de um gráfico de dispersão (gráficos "marginais") veja a página [Dicas para o ggplot]. 

### Gráficos: Violino, jitter, sina {.unnumbered}

> Below is code for creating **violin plots** (`geom_violin`) and **jitter plots** (`geom_jitter`) to show distributions. You can specify that the fill or color is also determined by the data, by inserting these options within `aes()`. 

Author: ```suggestion
Abaixo está o código para criar **gráficos de violino** (`geom_violin`) e **gráficos jitter** (`geom_jitter`) para mostrar as distribuições. Você pode especificar que o preenchimento ou a cor também são determinados pelos dados, inserindo estas opções dentro de `aes()`.. 

```

```{r fig.show='hold', message=FALSE, warning=FALSE, out.width=c('50%', '50%')}

> # A) Jitter plot por grupo

Author: ```suggestion
# A) Gráfico de jitter por grupo

```

```{r, echo = F}
outcomes <- linelist %>% 
  drop_na() %>% 
  group_by(outcome) %>% 
  count %>% 
  ungroup() %>% # desagrupar para que proporção esteja fora do total
  mutate(`proporcao` = n/sum(n)*100) # Calcular a porcentagem
  
outcomes #Visualizar a tabela completa
```

Abaixo está o código utilizando `geom_col` para criar gráficos de barras simples para mostrar a distribuição dos resultados dos pacientes com Ebola. Com o geom_col, tanto x como y precisam ser especificados. Aqui x é a variável categórica ao longo do eixo x, e y é a coluna de proporções geradas `proporcao`.  

```{r, fig.height = 3, fig.width=4.5}
# Desfecho em todos os casos
ggplot(outcomes) + 
  geom_col(aes(x=outcome, y = proportion)) +
  labs(subtitle = "Número de casos recuperados e óbitos de Ebola")

```

Para mostrarmos as detalhamentos por hospital, precisaríamos que nossa tabela contivesse mais informações e que estivesse em formato "longo". Criamos esta tabela com as freqüências das categorias combinadas `outcome` (Desfecho) e  `hospital` (ver página [Agrupando dados] para dicas de agrupamento). 

```{r, fig.height = 4, fig.width=6}
outcomes2 <- linelist %>% 
  drop_na(outcome) %>% 

>   count(hospital, outcome) %>%  # abtem contagens para hospital e desfecho 

Author: ```suggestion
  count(hospital, outcome) %>%  # obtém contagens para hospital e desfecho 

```

  group_by(hospital) %>%        # Agrupa para que proporção esteja fora do total do hospital 

>   mutate(proportion = n/sum(n)*100) # Calcule as proporções

Author: ```suggestion
  mutate(proportion = n/sum(n)*100) # Calcula as proporções

```
